routers/repo/issue_dependency_add.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package repo
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/context"
+)
+
+// AddDependency adds new dependencies
+func AddDependency(c *context.Context) {
+
+	depID, err := strconv.ParseInt(c.Req.PostForm.Get("newDependency"), 10, 64)
+	if err != nil {
+		c.Handle(http.StatusBadRequest, "issue ID is not int", err)
+		return
+	}
+
+	issueIndex := c.ParamsInt64("index")
+	issue, err := models.GetIssueByIndex(c.Repo.Repository.ID, issueIndex)
+	if err != nil {
+		c.Handle(http.StatusInternalServerError, "GetIssueByIndex", err)
+		return
+	}
+
+	// Check if the Repo is allowed to have dependencies
+	if !c.Repo.CanUseDependencies(issue, c.User) {
+		c.Handle(404, "MustEnableIssueDependencies", nil)
```

Comment:
Wrong error message

Reply:
Changed it to something more appropriate.